File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/PayRollCycleDto.cs
using System.Text.Json.Serialization;
using ProjectPos.Data.Shared.Enums;

namespace ProjectPos.Services.DTOs;

public class PayRollCycleDto
{
    public int Month { get; set; }
    public int Year { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IsClosed { get; set; } = false;
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public PayRollStatus PayRollStatus { get; set; } = PayRollStatus.Pending;

    public List<PaySlipDto>? PaySlips { get; set; }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/payroll-cycle-dto.ts
import { PayRollStatus } from "./payroll-status";
import { PaySlipDto } from "./payslip-dto";

export interface PayRollCycleDto {
    id?: number;
    month?: number;
    year?: number;
    startDate?: Date;
    endDate?: Date;
    isClosed?: boolean;
    payRollStatus?: PayRollStatus;
    paySlips?: PaySlipDto[];
} 